metric_config_parser/metric.py

```diff
@@ -66,14 +66,15 @@ class Metric:
     """
 
     name: str
-    data_source: DataSource
-    select_expression: str
+    data_source: Optional[DataSource]
+    select_expression: Optional[str]
```

Comment:
Should these default to `None`?

Reply:
Oh I see you're explicitly setting them to `None` for upstream metrics, that seems sufficient to me.